components/builder-worker/src/runner/util.rs

```diff
@@ -241,3 +235,18 @@ pub fn docker_exporter_spec(workspace: &Workspace) -> DockerExporterSpec {
         custom_tag: custom_tag,
     }
 }
+
+fn get_optional_args(opts: &JsonValue, arg: String) -> Option<String> {
+    let is_string = format!("{} value is a string", arg);
+    match opts.get(arg) {
+        Some(key) => {
+            let key = key.as_str().expect(&is_string);
```

Comment:
You might want to return a `Result<Option<String>>` here if there's a possibility that the given value isn't a string; calling `expect` will panic.

Alternatively, you might log the error and just return `None` here.